plasma_framework/test/src/exits/payment/PaymentChallengeIFENotCanonical.test.js

```diff
@@ -28,42 +31,115 @@ contract('PaymentInFlightExitRouter', ([_, ifeOwner, inputOwner, outputOwner, co
     const INITIAL_IMMUNE_EXIT_GAME_NUM = 1;
     const OUTPUT_TYPE_ONE = 1;
     const IFE_TX_TYPE = 1;
-    const BLOCK_NUMBER = 1000;
-    const AMOUNT = 10;
     const YOUNGEST_POSITION_BLOCK = 1000;
     const INFLIGHT_EXIT_YOUNGEST_INPUT_POSITION = buildUtxoPos(YOUNGEST_POSITION_BLOCK, 0, 0);
     const ETH = constants.ZERO_ADDRESS;
+    const DUMMY_OUTPUT_ID_FOR_OUTPUT = web3.utils.sha3('dummy output id for output');
+    const TEST_IFE_INPUT_AMOUNT = 999;
+    const TEST_IFE_OUTPUT_AMOUNT = 990;
+    const TEST_COMPETING_TX_OUTPUT_AMOUNT = 990;
+    const INPUT_TX_BLOCK_NUM = 1000;
+    const INPUT_UTXO_POS = new UtxoPos(buildUtxoPos(INPUT_TX_BLOCK_NUM, 0, 0));
+    const COMPETING_TX_BLOCK_NUM = 2000;
+    const DUMMY_OUTPUT_GUARD = web3.utils.utf8ToHex('dummy output guard for shared input');
+    const DUMMY_CONFIRM_SIG = web3.utils.utf8ToHex('dummy confirm sig for shared input');
+    const DUMMY_SPENDING_CONDITION_OPTIONAL_ARGS = web3.utils.utf8ToHex('dummy spending condition optional args');
+
+    const createInputTransaction = () => {
+        const output = new PaymentTransactionOutput(TEST_IFE_INPUT_AMOUNT, inputOwner, ETH);
+        const inputTx = new PaymentTransaction(
+            IFE_TX_TYPE, [buildUtxoPos(0, 0, 0)], [output],
+        );
+
+        return {
+            inputTx: web3.utils.bytesToHex(inputTx.rlpEncoded()),
+            decodedInputTx: inputTx,
+        };
+    };
+
+    const createCompetitorTransaction = () => {
+        const output = new PaymentTransactionOutput(
+            TEST_COMPETING_TX_OUTPUT_AMOUNT, buildOutputGuard(OUTPUT_TYPE_ONE, competitorOwner), ETH,
+        );
+        const competingTx = new PaymentTransaction(IFE_TX_TYPE, [INPUT_UTXO_POS.utxoPos], [output]);
+        const competingTxPos = new UtxoPos(buildUtxoPos(COMPETING_TX_BLOCK_NUM, 0, 0));
 
-    const buildInFlightExitData = async (exitIdHelper) => {
+        return {
+            competingTx: web3.utils.bytesToHex(competingTx.rlpEncoded()),
+            decodedCompetingTx: competingTx,
+            competingTxPos,
+        };
+    };
+
+    const buildValidNoncanonicalChallengeArgs = (decodedIfeTx) => {
```

Comment:
I move the function back to this file instead of `ife.js` so that we can use the test setup global variable for this test. Alternative would be injecting with function args instead but it would be slightly more complex to write.